### ISSUE TITLE
Lets bring The power of Pascal and Pandroid bundle to yours wonderful tool

### ISSUE THREAD

**Issue by JanuszChmiel** [SDRausty/buildAPKs#23]
Dear Mr Rausty,
Lets allow Pascal users to create Pascal applications directly on Android devices by using PPCJVM, which will create Java classes and Apache Ant will create .dex files by using special Android SDK tool. This bundle is also capable to create truly native ARM or ARM64 Bit .so libraryes.
The author of The bundle use his bundle to create fully functioning apps for his programmers work.
What would be needed?
I Am afraid, that we will not have The chance to bring fully functioning Oracle Java Open JDK which will work for Termux. Editing source code of such complex tool would be too complex for AN programmers wolunteers around The world.
And I do not think, that Apache Ant will work without Java Open JDK.
We must use only Open JDK, because it is not legal to create Android apps even for personal purposes by using non Open Java JDK. This is also The reason why Android studio is shipped with Open JDK.
This project will not require graphical apps. My goal is to avoid using of Lazarus IDE. SOme of its parts do not support Proot environment. Its build in macro interpreter.
We only can make compilation script which will call FPC compiler. So I suggest you to work together by using yours excellent ArchLinux script. So We would had to use ArchLinux as A work environment.
No clean Termux.
Existing Open JDK can hang and can frooze The device. I Am writing about Open JDK for Termux.
Pandroid bundle is here:

https://github.com/zeljkoc/pandroid
Thank you for yours support.

**Comment by JanuszChmiel** [comment]
Pandroid app is little Gui based Pascall app, which can create some template, code template. There are many provided Pascal examples in The example folder.
I Am sending you A sample Bash script in attachment. So you will get imagination how The compiler compiles The project and how .apk package is being created.
[build_debug_apk.txt](https://github.com/SDRausty/buildAPKs/files/4197379/build_debug_apk.txt)

I think, that we have big hope, if we would use ArchLinux directly. I only hope, that ArchLinux packagers have included OpenJDK to their repositories.
So I will remove Ubuntu 19.04, I will install ArchLinux by using yours professionally prepared script, which is being instensively developed. FOrtunately, Openjdk do not use shared memory. I have even been able to run Gradle on Android device.

**Comment by JanuszChmiel** [comment]
The most complex mission would be to manipulate with folders. Pandroid have been primarily created for Linux environment, but for 64 Bit Linux OS. So it would be required to move. FPC related modules from FPC installation for ARM64 bit edition to The corresponding folder. The Lazarus bundle would be perfect if it would work, but I do not assume, that somebody would have so much time to recode its build in macro interpreter to be compatible with Proot environment without sharing memory support.
This is The reason why want to only make console based approach. Based on Bash script interpretation.

**Comment by SDRausty** [comment]
After searching https://github.com/termux/termux-packages/issues?q=pascal I was surprised at the fact that Termux does not have a pascal compiler in the main packages repository.  

> I already have a fp-compiler
> Read readme.txt https://github.com/its-pointless/gcc_termux from there.

reference https://github.com/termux/termux-packages/pull/3558 free pascal packages

There is no information about https://github.com/termux/termux-packages/issues?q=pandroid so you may wish to open a https://github.com/termux/termux-packages/blob/master/.github/ISSUE_TEMPLATE/package_request.md package request at https://github.com/termux/termux-packages/issues  Hopefully, it will be easy to add.

**Comment by JanuszChmiel** [comment]
Thank you very much for yours kind answer.
Pandroid contain FPC or PPCJVM. PPCJVM can generate Java Class files
compatible with Java for desktop or for Android.
Pandroid bundle is very unique project for visually impaired programmers.
User can create even complex GUI without having imagination about Pixel
values or DDI or DIP for individual objects.
Next advantage is, that bundle support calling native .so libraries.
Author have even created special Pascal app which is capable to help
advanced programmers to generate Android Pascal APi header files. Project
contain Android 27 API. Generation of Android APi header is not fully
automatic process and it is not easy.
The Lamw Lamw module wizard is good for sighted majority. Developer tailor
window by using form designer and developer see The GUi elements on The
screen when position values are being changed.
Unfortunately, Lamw is unusable for visually impaireds, if somebody would
not only want to create apps which will only use keyboard shortcuts and TTS
engines or voice input.

Pandroid has allowed Me to create some simple apps with more than 20
buttons. This is The reason why I like PPandroid project so much.

**Comment by SDRausty** [comment]
> Pandroid contain FPC

yes, but as *.so s

**Comment by SDRausty** [comment]
> Pandroid has allowed Me to create some simple apps with more than 20
buttons.

: clapping hands :+1:

**Comment by JanuszChmiel** [comment]
aYes. Pandroid contin 64 Bit versions of FPC and Lazarus Ide. No ARM64
variant. Arch Linux standarnd repositoryes do not contain Lazarus ide.
I have cooperated deeply with The auhor of Pandroid. I have been able to
create simple Internet radio plazer which play all Czech state radio
stations by using Bass.so library.
Sure. It is nothing fancy for sighted professionals, but for Me, it was very
positive experience. Because GUI have been usable even on Android 9 and I
could create it with no need to type numbers to specifi width and height for
every element.
I know, that my apps would not be very nice for sighted majority. But it so
nice feeling, that I can create app with no need to use RFO Basic and that I
do not have to bsad that Lamw is based on .lfm formulars with object
positions values.
Bu I know, that making Pandroid to be usable on Termux is too complex
mission.
It would require many file and folder moving operations and Ant will not
work directly from Termux.
But I will use Arch Linux because it is stable and reliable distribution and
their programmers are doing their best to make this system even for ARM64
CPU.

**Comment by JanuszChmiel** [comment]
I have decided to work directly from ArchLinux. This distribution is giving
Me every day new surprice. And even though some packages are not available
for Arch64 ARM architecture this stable and user friendly distro have even
to start my motivate to create ARM64 Bit packages. I will start with AUR pkg
format first and when I will learn about ArchLinux more deeper, I will do my
best to prepare Pacman compatible packages.
Mr Raustry, thank you very much that you have opened me The door to
outstanding world of Arch Linux.
It is every think thanks to yours script and thanks to Proot.
Android is giving Me secure base. I have compiled Gradio for ARM64 Bit
edition yesterday. It work perfectly. Now, I have even installed
Gnome-builder which speak with Orca!!! It is not common, since I could not
use Gnome-builder with Orca from Debian or Ubuntu while using Proot and
Android. But dbus-launch app name approach have much more positive results
than I could think before.
Sure. I have been A little engry for The first time. Typing dbus-launch
before many X11 apps names have been un common for Me. But now, I Am glad,
that Dbus work atleast bz using this technique. Mate have some issues mainly
various mate panels.
I have even decided that I will try to build newest stable Seamonkey for
ARM64 Bit architecture. Because community repo contain older release than
newest stable.
I must setup sudo because Mozila bootstrap script require it becausemakepkg
do not work from root ccount. Sure. I could modify The script and I could
remove condition and warning, but I think, that core developers have thinked
it well with The users that compiling package as root is not good technique.
To build Seamonkey, I will use technical information about compiler setup
ETC and I will try to use it for build The newest version.
If build will crash thanks to little available memory, I will have to use
some standard PC and I will had to switch to ARM64 Bit emulator. I hope,
that Fedora and VIrt manager will allow Me to do that.
May be, that build of Seamonkey will be so long process, that it will
produce too much heat of The phone board so I will have to stop it to
prevent phone overburn and corruption.
Or I will receive The series of memory allocation errors.
It will be exciting informatic adventure. I do not know where to buy some
ARM64 BIT compatible computer with ARM64 CPU.
